jep/401/README.adoc

```diff
@@ -0,0 +1,245 @@
+
+By consolidating all permission types that effectively allow a user to have full `root` access on a Jenkins instance, and introducing a new permission (`Jenkins.CONFIGURE`) that allows a limited amount of access to configure certain non-critical functionality, a Jenkins administrator can safely delegate configuration aspects while being confident that security concerns are being met.
+
+The `Jenkins.CONFIGURE` permission type is not intended to replace any `Item` level permissions.  It is solely for the purpose of seperating sensistive, security related Jenkins configuration settings from those that a non-root administrator might be expected to manage.
+
+== Reasoning
+
+[TIP]
+====
+Explain why particular design decisions were made.
+Describe alternate designs that were considered and related work. For example, how the feature is supported in other systems.
+Provide evidence of consensus within the community and discuss important objections or concerns raised during discussion.
+
+* Use sub-headings to organize this section for ease of readability.
+* Do not talk about history or why this needs to be done - that is part of Motivation section.
+====
+
+=== Consolidating "dangerous" permissions
+The permissions that are considered "dangerous" effectively provide the user with the means to grant themselves the "ultimate" permission, `Jenkins.ADMINISTER`, thereby making their existence redundant.  Consolidating these permissions into `Jenkins.ADMINISTER` removes ambiguity concerning what a user is or is not permitted to do.
+
+=== Deprecating "dangerous" permissions vs. removing them
+Due to the large number of plugins available for Jenkins, the decision was made to mark these "dangerous" permissions as `@Deprecated` and `@Restricted(NoExternalUse.class)`.  This will allow for backwards compatibility with existing plugins while minimizing any new usages.  This approach will encourage plugin authors to update their code in future releases that depend upon the new Jenkins Core baseline.
+
+== Backwards Compatibility
+
+[TIP]
+====
+Describe any incompatibilities and their severity.
+Describe how the JEP proposes to deal with these incompatibilities.
+
+If there are no backwards compatibility concerns, this section may simply say:
+There are no backwards compatibility concerns related to this proposal.
+====
+No existing permission types are being removed at this point, which will allow for existing plugins that make use of them to continue to function.  Removing the "dangerous" permissions is outside the scope of this JEP.
```

Comment:
Same question than above. The permission are not removed but actions will be requiring ADMINISTER permission....so compatibility will break?

Reply:
Based on my understanding, we will not be breaking compatibility because anyone with `Jenkins.RUN_SCRIPTS` already has `Jenkins.ADMINISTER`.  See [SECURITY-410](https://jenkins.io/security/advisory/2017-04-10/#matrix-authorization-strategy-plugin-allowed-configuring-dangerous-permissions) for details.

@aHenryJard / @EstherAF - can you please sanity check me?

Reply:
If I understand [this](https://github.com/jenkinsci/matrix-auth-plugin/commit/246d7d744f7280978546fc67e643f2c950713a87) correctly, it means any call to `hasPermission(Jenkins.RUN_SCRIPTS)` is effectively checking for `Jenkins.ADMINISTER`. So they are already requiring it under the hood. So this wouldn't break anything.